sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample8_ConsumeEventsByBatch.cs

```diff
@@ -0,0 +1,135 @@
+        ///
+        public string Name { get; } = nameof(Sample8_ConsumeEventsByBatch);
+
+        /// <summary>
+        ///   A short description of the sample.
+        /// </summary>
+        ///
+        public string Description { get; } = "An example of consuming events, using a batch approach tocontrol throughput.";
+
+        /// <summary>
+        ///   Runs the sample using the specified Event Hubs connection information.
+        /// </summary>
+        ///
+        /// <param name="connectionString">The connection string for the Event Hubs namespace that the sample should target.</param>
+        /// <param name="eventHubName">The name of the Event Hub, sometimes known as its path, that she sample should run against.</param>
+        ///
+        public async Task RunAsync(string connectionString,
+                                   string eventHubName)
+        {
+            // We will start by creating a client each using its default set of options.
+
+            await using (var client = new EventHubClient(connectionString, eventHubName))
+            {
+                // With our client, we can now inspect the partitions and find the identifier
+                // of the first.
+
+                string firstPartition = (await client.GetPartitionIdsAsync()).First();
+
+                // In this example, we will create our consumer for the first partition in the Event Hub, using the default consumer group
+                // that is created with an Event Hub.  Our consumer will begin watching the partition at the very end, reading only new events
+                // that we will publish for it.
+
+                await using (EventHubConsumer consumer = client.CreateConsumer(EventHubConsumer.DefaultConsumerGroup, firstPartition, EventPosition.Latest))
+                await using (EventHubProducer producer = client.CreateProducer(new EventHubProducerOptions { PartitionId = firstPartition }))
+                {
+                    // Because our consumer is reading from the latest position, it won't see events that have previously
+                    // been published. Before we can publish the events, we will need to ask the consumer to perform an operation,
+                    // because it opens its connection only when it needs to.  The first receive that we ask of it will not see
+                    // any events, but will allow the consumer to start watching the partition.
+
+                    await consumer.ReceiveAsync(1, TimeSpan.Zero);
+
+                    // Now that the consumer is watching the partition, let's publish a fair sized batch of events
+                    // we would like for it to consume.
+
+                    int eventBatchSize = 100;
+
+                    IEnumerable<EventData> eventBatch =
+                        Enumerable
+                          .Range(0, eventBatchSize)
+                          .Select(index => new EventData(Encoding.UTF8.GetBytes($"I am event #{ index }")));
```

Comment:
I would find a for-loop over an array easier to parse. This seems too cute. :-)

Reply:
Not sure that I agree, given how long Linq and the related constructs have been with us, but I'm not of strong opinion and would prefer to err on the side of caution.   Refactoring to an array/for-loop.